src/server.c

```diff
@@ -3371,6 +3371,16 @@ int processCommand(client *c) {
         return C_ERR;
     }
 
+    /* If we're inside a module blocked context yielding that wants to avoid
+     * processing clients, postpone the command. */
+    if (server.busy_module_yield_flags!=BUSY_MODULE_YIELD_NONE &&
+        !(server.busy_module_yield_flags&BUSY_MODULE_YIELD_CLIENTS))
+    {
+        c->bpop.timeout = 0;
+        blockClient(c,BLOCKED_PAUSE);
```

Comment:
I agree about renaming this BLOCKED_POSTPONE. There are several places that explicitly make reference to the fact this is only called in a paused context, included the documentation for it. I don't see any specific issues as long as we believe calling the yield is an infrequent event.

Reply:
calling the yield is not infrequent, but i do "un pause" them only when the outer most module context is done.
so the other thing we can do is we throttle (delay) the initial yielding.

assuming the module calls us very frequently, currently there are 3 modes of operation:
1. during loading, in which case we start processing events right away (in HZ intervals, so the first one is after some 100ms of blockage). in this mode we always process all clients (return -LOADING), so it won't postpone any
2. during other blocked contexts, when the user used YIELD_CLIENTS, we start processing event only after 5 seconds of blockage, and also we don't postpone any (return -BUSY to all).
3. during other blocked contexts, when the user used YIELD_EVENTS. discuss this below.

in this mode we could have ignored the `busy_reply_threshold` config, and start yielding right away (after 100ms hz delay), in which case we could do many blocking and unblocking of clients.
i didn't do that (we do wait the 5 seconds) because i was afraid of side effects such as this one on short blocking operations.

in theory, even if the user used YIELD_CLIENTS where we start yielding to clients only after the 5 seconds threshold, we could have stated yielding to other events sooner (i.e. right away after the hz delay).
again, decided not to do that since it seemed risky.

the meaning of that is that if there are any bugs, we're unlikely to see them. i.e they'll surface only when the module blocks for more than 5 seconds.. if we used the shorter throttling, maybe it would have exposed bugs sooner (during release candidate testing).

WDYT?

Reply:
added a commit with a comment to explain that, so we won't overlook this in the future.

Reply:
I think the tradeoffs made are ok.